README.md

```diff
@@ -126,3 +126,63 @@ in performance.
 
 Consult the [Getting Started](docs/GettingStarted.md) page for detailed 
 information on configuring and compiling CIRCT.
+
+## Using the Python Bindings
+
+If you are mainly interested in using CIRCT from Python scripts, you need to compile both LLVM/MLIR and CIRCT with Python bindings enabled. To do this, follow the steps in "Setting things up" above, adding the following options:
+
+```
+# in the LLVM/MLIR build directory
+cmake [...] -DMLIR_BINDINGS_PYTHON_ENABLED=ON
+
+# in the CIRCT build directory
+cmake [...] -DCIRCT_BINDINGS_PYTHON_ENABLED=ON
+```
+
+Afterwards, use `ninja check-circt-integration` to ensure that the bindings work. (This will now additionally spin up a couple of Python scripts to test that they are accessible.)
+
+### Without Installation
+
+If you want to try the bindings fresh from the compiler without installation, you need to ensure Python can find both the additional modules as well as the compiled native libraries:
+
+```
+export PYTHONPATH="$PWD/llvm/build/python:$PWD/build/python:$PYTHONPATH"
+export LD_LIBRARY_PATH="$PWD/llvm/build/lib:$PWD/build/lib:$LD_LIBRARY_PATH"
+```
+
+### With Installation
+
+If you are installing CIRCT through `ninja install` anyway, the libraries and Python modules will be installed into the correct location automatically.
+
+### Trying things out
+
+Now you are able to use the CIRCT dialects and infrastructure from a Python interpreter and script:
+
+```python
+# silicon.py
+import circt
+import mlir
+from mlir.ir import *
+from circt.dialects import hw, comb
+
+with Context() as ctx, Location.unknown():
+  circt.register_dialects(ctx)
+  i42 = IntegerType.get_signless(42)
+  m = Module.create()
+  with InsertionPoint(m.body):
+    @hw.HWModuleOp.from_py_func(i42, i42)
```

Comment:
This decorator is kinda nice for toy examples, but we have developed a [more powerful](https://github.com/llvm/circt/blob/main/lib/Bindings/Python/circt/dialects/_hw_ops_ext.py#L10) way to instantiate modules, which this will not work with (output port names must be named, which this decorator doesn't do). We were planning to yank this out, and @teqdruid is working on a replacement that decorates a class. This is just an FYI that when that happens we would need to update this.

Reply:
That's good to know! Would be great if we can update this as well once the `from_py_func` goes. Just some minimal example will do I think.